egs/wsj_mix/asr1/run.sh

```diff
@@ -0,0 +1,382 @@
+debugmode=1
+dumpdir=dump   # directory to dump full features
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0      # verbose option
+resume=        # Resume the training from snapshot
+seed=1
+
+# feature configuration
+do_delta=true
+
+# network architecture
+# encoder related
+etype=vggblstmp     # encoder architecture type
+elayers_sd=1
+elayers_rec=2
+eunits=1024
+eprojs=1024
+subsample=1_2_2_1_1 # skip every n frame from input to nth layers
+num_spkrs=2         # number of speakers
+# decoder related
+dlayers=1
+dunits=300
+# attention related
+atype=location
+spa=false           # speaker parallel attention
+adim=320
+awin=5
+aheads=4
+aconv_chans=10
+aconv_filts=100
+
+# hybrid CTC/attention
+mtlalpha=0.2
+
+# label smoothing
+lsm_type=unigram
+lsm_weight=0.05
+
+# minibatch related
+batchsize=10
+maxlen_in=800  # if input length  > maxlen_in, batchsize is automatically reduced
+maxlen_out=150 # if output length > maxlen_out, batchsize is automatically reduced
+
+# optimization related
+opt=adadelta
+epochs=15
+patience=3
+
+# rnnlm related
+use_wordlm=true     # false means to train/use a character LM
+lm_vocabsize=65000  # effective only for word LMs
+lm_layers=1         # 2 for character LMs
+lm_units=1000       # 650 for character LMs
+lm_opt=sgd          # adam for character LMs
+lm_batchsize=300    # 1024 for character LMs
+lm_epochs=20        # number of epochs
+lm_patience=3
+lm_maxlen=40        # 150 for character LMs
+lm_resume=          # specify a snapshot file to resume LM training
+lmtag=              # tag for managing LMs
+
+# decoding parameter
+lm_weight=1.0
+beam_size=30
+penalty=0.0
+maxlenratio=0.0
+minlenratio=0.0
+ctc_weight=0.3
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+
+# scheduled sampling option
+samp_prob=0.0
+
+# data
+wsj0=/export/corpora5/LDC/LDC93S6B
+wsj1=/export/corpora5/LDC/LDC94S13B
+wsj_full_wav=$PWD/data/wsj0
+wsj_2mix_wav=$PWD/data/local/wsj_mix/2speakers
+wsj_2mix_scripts=$PWD/data/local/wsj_mix/scripts
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+. ./path.sh
+. ./cmd.sh
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=tr
+train_dev=cv
+recog_set=tt
+
+if [ ${stage} -le 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    echo "stage 0: Data preparation"
+    ### This part is for WSJ0 mix
+    ### Download mixture scripts and create mixtures for 2 speakers
+    local/wsj0_create_mixture.sh ${wsj_2mix_scripts} ${wsj_full_wav} \
+        ${wsj_2mix_wav} || exit 1;
+    local/wsj0_2mix_data_prep.sh ${wsj_2mix_wav}/wav16k/max ${wsj_2mix_scripts} \
+        ${wsj_full_wav} || exit 1;
+
+    ### Or this part is for WSJ mix
```

Comment:
I think it is another way of data preparation when somebody wants to create the mixture from wsj_si284, rather than WSJ0. If we want to make our results in the ICASSP paper reproducible, maybe we should keep this.